huaweicloud/resource_huaweicloud_identity_acl.go

```diff
@@ -0,0 +1,226 @@
+						},
+					},
+				},
+			},
+			"ip_ranges": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"range": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateIPRange,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIdentityACLCreate(d *schema.ResourceData, meta interface{}) error {
+	updateACLPolicy(d, meta)
+
+	d.SetId(meta.(*Config).DomainID)
+	return resourceIdentityACLRead(d, meta)
+}
+
+func resourceIdentityACLRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	iamClient, err := config.IAMV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud iam client: %s", err)
+	}
+	domainID := config.DomainID
+	if domainID == "" {
+		return fmt.Errorf("the domain_id must be specified in the provider configuration")
+	}
+
+	var identityACL *acl.ACLPolicy
+	switch d.Get("acl_type").(string) {
+	case "console":
+		res, err := acl.ConsoleACLPolicyGet(iamClient, domainID).ConsoleExtract()
+		identityACL = res
+		if err != nil {
+			return fmt.Errorf("Error fetching identity acl for console access")
+		}
+		log.Printf("[DEBUG] Retrieved HuaweiCloud identity acl: %#v", identityACL)
+	case "api":
+		acl, err := acl.APIACLPolicyGet(iamClient, domainID).APIExtract()
+		identityACL = acl
+		if err != nil {
+			return fmt.Errorf("Error fetching identity acl for api access")
+		}
+		log.Printf("[DEBUG] Retrieved HuaweiCloud identity acl: %#v", identityACL)
+	}
+
+	if len(identityACL.AllowAddressNetmasks) > 0 {
+		addressNetmasks := make([]map[string]string, 0, len(identityACL.AllowAddressNetmasks))
+		for _, v := range identityACL.AllowAddressNetmasks {
+			addressNetmask := map[string]string{
+				"cidr":        v.AddressNetmask,
+				"description": v.Description,
+			}
+			addressNetmasks = append(addressNetmasks, addressNetmask)
+		}
+		d.Set("ip_cidrs", addressNetmasks)
+	}
+	if len(identityACL.AllowIPRanges) > 0 {
+		ipRanges := make([]map[string]string, 0, len(identityACL.AllowIPRanges))
+		for _, v := range identityACL.AllowIPRanges {
+			ipRange := map[string]string{
+				"range":       v.IPRange,
+				"description": v.Description,
+			}
+			ipRanges = append(ipRanges, ipRange)
+		}
+		d.Set("ip_ranges", ipRanges)
+	}
+
+	return nil
+}
+
+func resourceIdentityACLUpdate(d *schema.ResourceData, meta interface{}) error {
+	if d.HasChanges("ip_cidrs", "ip_ranges") {
+		updateACLPolicy(d, meta)
+	}
+
+	return resourceIdentityACLRead(d, meta)
+}
+
+func resourceIdentityACLDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	iamClient, err := config.IAMV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud iam client: %s", err)
+	}
+	domainID := config.DomainID
```

Comment:
using d.Id() will be better

Reply:
Thanks, now using d.Id() to get ACL id and assigned to domain Id.